modules/runner/src/test/java/org/apache/ignite/internal/configuration/compatibility/ConfigurationCompatibilityTest.java

```diff
@@ -0,0 +1,174 @@
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.configuration.compatibility;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.ignite.configuration.ConfigurationModule;
+import org.apache.ignite.configuration.RootKey;
+import org.apache.ignite.internal.configuration.ConfigurationModules;
+import org.apache.ignite.internal.configuration.ServiceLoaderModulesProvider;
+import org.apache.ignite.internal.configuration.compatibility.framework.ConfigNode;
+import org.apache.ignite.internal.configuration.compatibility.framework.ConfigNodeSerializer;
+import org.apache.ignite.internal.configuration.compatibility.framework.ConfigShuttle;
+import org.apache.ignite.internal.configuration.compatibility.framework.ConfigurationSnapshotManager;
+import org.apache.ignite.internal.configuration.compatibility.framework.ConfigurationTreeComparator;
+import org.apache.ignite.internal.configuration.compatibility.framework.ConfigurationTreeScanner;
+import org.apache.ignite.internal.configuration.compatibility.framework.ConfigurationTreeScanner.ScanContext;
+import org.apache.ignite.internal.logger.IgniteLogger;
+import org.apache.ignite.internal.logger.Loggers;
+import org.apache.ignite.internal.testframework.IgniteAbstractTest;
+import org.apache.ignite.internal.util.io.IgniteUnsafeDataInput;
+import org.apache.ignite.internal.util.io.IgniteUnsafeDataOutput;
+import org.jetbrains.annotations.Nullable;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+/**
+ * Tests for configuration compatibility.
+ */
+public class ConfigurationCompatibilityTest extends IgniteAbstractTest {
+    private static final String DEFAULT_FILE_NAME = "snapshot.bin";
+    private static final String SNAPSHOTS_DIRECTORY = "./src/test/resources/";
+    private static final String DEFAULT_SNAPSHOT_FILE = "./modules/runner/build/compatibility/configuration/" + DEFAULT_FILE_NAME;
```

Comment:
is it possible to avoid using `current dir` in relative path? I think, for binaries in resources we can leverage `java.lang.ClassLoader#getResourceAsStream` (see `org.apache.ignite.internal.sql.engine.util.tpch.TpchHelper#loadFromResource` for reference). And to put intermediate binaries created during test we usually use concept of `workDir` (see `org.apache.ignite.internal.testframework.WorkDirectoryExtension`).

Beside, will these paths work on other platforms? I fill like there may be problems on Windows since they use `\` as separator